### DISCUSSION TITLE
Heading Group

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#323] [Component suggestions]
<img width="480" alt="verschillende vormen van de heading-group component" src="https://github.com/user-attachments/assets/2cef1bd1-d2ac-4f82-9804-5fd8a7a78510">

We willen graag meer leren over de Heading Group component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1727875317001389)

---

## Naam

Heading Group

## Doel

Combinatie van een koptekst en ondersteunende tekst.

## Gerelateerde componenten
[Heading](https://github.com/orgs/nl-design-system/discussions/220), [Rich Text Content](https://github.com/orgs/nl-design-system/discussions/289), [Subheading](https://github.com/orgs/nl-design-system/discussions/292), [Pre-heading](https://github.com/orgs/nl-design-system/discussions/286)

## Onderzoek

[`<hgroup>`: The Heading Group element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hgroup)
Documentatie van Mozilla Developers Network over het `<hgroup>` HTML element.

[Accessible heading structure - The a11y project](https://www.a11yproject.com/posts/how-to-accessible-heading-structure/#hgroup)
Artikel door Rian Rietveld waarin ook wordt geschreven over het gebruik van `hgroup`. Rian geeft ook een voorbeeld in code van het toegankelijk opzetten van een Heading Group.

[Geef aan hoeveel stappen er zijn en in welke stap de gebruiker zich bevindt - NL Design System](https://www.nldesignsystem.nl/richtlijnen/formulieren/meerdere-stappen/voortgang-tonen)
Richtlijnen van NL Design System rondom het tonen van voortgang binnen een meerstappenformulier. Ook hier wordt een voorbeeld gegeven waarbij gebruik wordt gemaakt van `<hgroup>`.

**Comment by jeffreylauwers** [comment]
## DSO

Heading Group niet als component gevonden. Maar de 'Application Heading' component komt in de buurt.

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/voorbeeldpagina-s-toepassingen-componenten-application-heading-application-heading-met-subtitle-en-step--application-heading-met-subtitle-en-step)

---

### Screenshots / Varianten

**Storybook**

Application Heading met Subtitle en Step
![Scherm­afbeelding 2024-10-02 om 11 22 10](https://github.com/user-attachments/assets/a84d2de4-e5b1-425f-871f-75c8cdf60d5f)

---

### Notities

Combinatie van `h1` met `h2`. De `h2` bevat een `span`. De `span` heeft visueel een andere stijling.

**Comment by jeffreylauwers** [comment]
## DSO

Heading Group niet als component gevonden. Maar de tekstopmaak binnen de 'Hero' component komt in de buurt.

[Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-hero--documentatie)

---

### Screenshots / Varianten

**Storybook**

Hero
![Scherm­afbeelding 2024-10-02 om 11 35 30](https://github.com/user-attachments/assets/7d4576ae-ca0f-42bc-a926-ad2a79a862e8)

---

### Notities

`h1` met daarbinnen een `span`. De `span` heeft visueel een andere stijling.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-heading-group--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/heading-group)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-10-02 om 11 38 34](https://github.com/user-attachments/assets/e6b7d4e8-5a8c-40e4-8721-5156ef08cb8f)

Pre-heading
![Scherm­afbeelding 2024-10-02 om 11 38 55](https://github.com/user-attachments/assets/01ed1e02-2710-4ce7-8e98-911d3a7e30e9)

---

### Notities

- Documentatie beschikbaar.
- `<hgroup>` HTML element met daarbinnen een `h1` en een `p`.
- Bij de 'pre-heading' variant komt in code de 'p' na de 'h1'. Visueel wordt de `p` dan boven de `h1` getoond. Hierdoor mist men geen informatie wanneer zij via Headings navigeren.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Heading Group op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=heading+group)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Heading Group op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=heading+group)
